src/simulation/hdf_simulation_store.jl

```diff
@@ -734,18 +767,55 @@ function _read_data_columns(
     return data, columns
 end
 
+function _read_data_columns(
+    store::HdfSimulationStore,
+    model_name::Symbol,
+    key::OptimizationContainerKey,
+    index::EmulationModelIndexType,
+)
+    if is_cached(store.cache, model_name, key, index)
+        # data = read_result(store.cache, model_name, key, index)
+        #columns = _get_em_dataset(store, model_name, key).column_dataset[:]
+    else
+        data, columns = _read_result(store, model_name, key, index)
+    end
+
+    return data, columns
```

Comment:
This looks like a bug - will crash if is_cached is true.

Reply:
Right now it is short circuited. I will comment the code in the mean time.